src/app/shared/components/size-picker/size-picker.component.ts

```diff
@@ -79,13 +28,25 @@ export class SizePickerComponent implements OnInit {
   }
 
   ngOnInit() {
-    while (this.sizes.length) {
-      this.sizeses.push(this.sizes.splice(0, 5));
-    }
+
   }
 
-  onChange(e) {
+  onchange(e) {
     this.value.emit(+e.value);
     this.val = +e.value;
   }
+
+  ngOnChanges(changes: SimpleChanges) {
+
+    const sizes: SimpleChange = changes.sizes;
+
+    this.sizeses = [];
+    if (changes.sizes.currentValue.length > 0) {
+      let _sizeArr = changes.sizes.currentValue;
+      for (let i = 0; i < _sizeArr.length; i++) {
+        this.sizeses.push(_sizeArr.splice(0, 5))
```

Comment:
suggestion : it's better that sizes array push with this model( maybe in other commits ):
 {
    name : farsi-numbers,
    value : english nums
}
,